girder/models/file.py

```diff
@@ -93,19 +94,42 @@ def download(self, file, offset=0, headers=True, endByte=None,
         :type contentDisposition: str or None
         :type extraParameters: str or None
         """
+        if offset is 0:
```

Comment:
`is` should be used to compare `None`, and to see if a value is explicitly `True` or `False`. Otherwise, use `==` to compare, as `is` looks at reference (location in memory) equality, not value equality.

This bug reveals that this code path is not adequately tested.